src/librustdoc/html/templates/page.html

```diff
@@ -7,8 +7,7 @@
     <meta name="description" content="{{page.description}}"> {# #}
     <title>{{page.title}}</title> {# #}
     <script>if(window.location.protocol!=="file:") {# Hack to skip preloading fonts locally - see #98769 #}
-    for(f of "{{files.source_serif_4_regular}},{{files.fira_sans_regular}},{{files.fira_sans_medium}},{{files.source_code_pro_regular}},{{files.source_code_pro_semibold}}".split(",")) {# #}
-     document.write(`<link rel="preload" as="font" type="font/woff2" crossorigin href="{{static_root_path|safe}}${f}">`) {# #}
+    document.head.insertAdjacentHTML("beforeend","{{files.source_serif_4_regular}},{{files.fira_sans_regular}},{{files.fira_sans_medium}},{{files.source_code_pro_regular}},{{files.source_code_pro_semibold}}".split(",").map(f=>`<link rel="preload" as="font" type="font/woff2" crossorigin href="{{static_root_path|safe}}${f}">`).join("")) {# #}
```

Comment:
```suggestion
    document.head.insertAdjacentHTML("beforeend",`"{{files.source_serif_4_regular}},{{files.fira_sans_regular}},{{files.fira_sans_medium}},{{files.source_code_pro_regular}},{{files.source_code_pro_semibold}}".split(",").map(f=><link rel="preload" as="font" type="font/woff2" crossorigin href="{{static_root_path|safe}}${f}">).join(""))` {# #}
```

It might have to be this, going by my expectation of what this template code does. But the `{# #}` is non-intuitive so I doubt my expectations.

What engine is it using so I can look at the docs?

Reply:
`{# #}` is comment in jinja. So it doesn't appear in the HTML. And the code you wrote is not doing the same thing. Currently, we want to generate a `<link>` for each font. With your change, we have a weird long template string which doesn't render what we want.

Reply:
What I'm expecting the generated javascript to look like is 

```javascript
document.head.insertAdjacentHTML("beforeend", `<link ...><link ...><link ...><link ...><link ...>`);
```

If that's already the case then everything is fine.

Reply:
Oh I see what you mean now. That'd be even better indeed!

Reply:
Ah wait no. The goal is to reduce the generated code size. So inlining it like this won't revert what we recently achieved. So no, we don't want it to handled when generating documentation.

Reply:
Oh I see, it's doing the concat in JS. I got confused what's evaluated by the template and what by javascript.
Yeah, makes sense now.